tests/60app-services/02ghost.pl

```diff
@@ -87,6 +87,72 @@
       })->then_done(1);
    };
 
+test "Application services can be not rate limited",
+   requires => [ as_ghost_fixture(), $main::AS_USER,
+                     room_fixture( requires_users => [ $user_fixture ] ),
+                qw( can_receive_room_message_locally )],
+
+   do => sub {
+      my ( $ghost, $as_user, $room_id ) = @_;
+
+      Future->needs_all(
+         ( map {
+            await_as_event( "m.room.member" )->then( sub { Future->done( 1 ); } ),
```

Comment:
There's a neater form here

    $f->then_done(1);

is a shortcut for

    $f->then( sub { Future->done(1) } );